lib/internal/modules/esm/utils.js

```diff
@@ -107,26 +108,19 @@ function isLoaderWorker() {
   return _isLoaderWorker;
 }
 
-async function initializeHooks() {
-  const customLoaderURLs = getOptionValue('--experimental-loader');
-
-  let cwd;
-  try {
-    // `process.cwd()` can fail if the parent directory is deleted while the process runs.
-    cwd = process.cwd() + '/';
-  } catch {
-    cwd = '/';
-  }
-
-
-  const { Hooks } = require('internal/modules/esm/hooks');
-  const hooks = new Hooks();
-
+const createHooksLoader = (hooks) => {
+  // TODO: HACK: `DefaultModuleLoader` depends on `getDefaultConditions` defined in
+  // this file so we have a circular reference going on. If that function was in
+  // it's on file we could just expose this class generically.
```

Comment:
I think this PR needs to wait until https://github.com/nodejs/node/pull/48439 lands, which refactors some of the things about how these classes interact.